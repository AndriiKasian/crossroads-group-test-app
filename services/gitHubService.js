import axios from '../utils/API';

/**
 * Load the commit list
 * @returns {Promise<AxiosResponse<T>>}
 */
export const getCommits = () => {
    return axios.get('/commits');
};
