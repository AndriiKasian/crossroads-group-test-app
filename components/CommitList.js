import React, {useEffect, useState} from "react";
import get from 'lodash.get';

import CommitListItem from "./CommitListItem";

import { getCommits } from '../services/gitHubService';

const CommitList = () => {
    const [list, setList] = useState([]);

    const loadCommitList = async () => {
        const res = await getCommits();
        const data = get(res, 'data') || [];

        setList(data)
    }

    useEffect(() => {
        loadCommitList()
    }, []);

    return (
        <div>
            {list.map(({ html_url: url, commit, sha }) => (
                <CommitListItem
                    key={sha}
                    committer={get(commit, 'committer.name')}
                    date={get(commit, 'committer.date')}
                    message={get(commit, 'message')}
                    url={url}
                />
            ))}
            {!list.length && <p>There is no commits to show</p>}
        </div>
    )
};

export default CommitList;
