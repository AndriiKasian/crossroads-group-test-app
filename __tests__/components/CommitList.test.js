import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'
import { act } from 'react-dom/test-utils';

import CommitList from '../../components/CommitList'
import * as gitHubService from '../../services/gitHubService'

describe('CommitList component', () => {
    it('should render with empty props', () => {
        const wrapper = shallow(<CommitList/>);

        expect(wrapper).toBeDefined();
    });

    it('should render correctly', () => {
        const component = renderer.create(<CommitList/>);
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should call loadCommitList method when mount', async () => {
        const spy = jest
            .spyOn(gitHubService, 'getCommits')
            .mockImplementation(() => {});

        await act(async () => mount(<CommitList/>));

        expect(spy).toHaveBeenCalledTimes(1);
    });

    afterEach(() => {
        jest.clearAllMocks()
    });
});
