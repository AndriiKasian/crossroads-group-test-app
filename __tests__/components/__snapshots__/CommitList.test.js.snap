// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitList component should render correctly 1`] = `<div />`;
